src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,324 @@
+import React, { Fragment, useCallback, useMemo, useEffect } from 'react'
+import styled from 'styled-components'
+import { th } from '@xstyled/system'
+import PropTypes from 'prop-types'
+import { Button, Icon } from '..'
+import { Flex as Container } from '../Grid'
+import Typography from '../Typography'
+
+const getLowerValue = (valueA, valueB) => {
+  if (valueA > valueB) {
+    return valueB
+  }
+  return valueA
+}
+
+const Pagination = ({
+  page,
+  pageSize,
+  onPageChange,
+  onPageSizeChange,
+  minPagesToShowDots,
+  pagesShownedBetweenDots,
+  variant,
+  ...props
+}) => {
+  useEffect(() => {
+    variant && (document.getElementById('page').value = page)
+  }, [page])
+
+  const onChangePageInput = () => {
+    setTimeout(
+      () =>
+        Number(document.getElementById('page').value) <= pageSize
+          ? setPage(Number(document.getElementById('page').value))
+          : setPage(page),
+      700
+    )
+  }
+
+  const setPage = useCallback(pageNumber => {
+    onPageChange && onPageChange(pageNumber)
+  }, [])
+
+  const getPages = useCallback(
+    state => {
+      const prevLength = getLowerValue(MIDDLE, page - 1)
+      const nextLength = getLowerValue(MIDDLE, pageSize - page)
+      if (state === 'prev') {
+        return pageSize > minPagesToShowDots ? getLowerValue(MIDDLE + MIDDLE - nextLength, page - 1) : page - 1
+      }
+
+      return pageSize > minPagesToShowDots
+        ? getLowerValue(MIDDLE + MIDDLE - prevLength, pageSize - page)
+        : pageSize - page
+    },
+    [page, pageSize]
+  )
+
+  const MIDDLE = useMemo(() => Math.trunc(pagesShownedBetweenDots / 2), [pagesShownedBetweenDots])
+  const prevLength = useMemo(() => getPages('prev'), [getPages])
+  const nextLength = useMemo(() => getPages('next'), [getPages])
+
+  const showFirstDots = page > 3 && pageSize > minPagesToShowDots
```

Comment:
isso vai funcionar se o `minPagesToShowDots` for menor que 7?

Reply:
Se for menor que 7 ele não deve mostrar as reticências.